app/src/androidTest/java/net/skyscanner/backpack/calendar/BpkCalendarTest.kt

```diff
@@ -0,0 +1,180 @@
+    get() = initialEndDate ?: super.endDate
+
+  override fun onRangeSelected(range: CalendarRange) {}
+
+  override fun isToday(year: Int, month: Int, day: Int): Boolean {
+    return day == 2 && month == 0 && year == 2019
+  }
+}
+
+@RunWith(AndroidJUnit4::class)
+class BpkCalendarTest : BpkSnapshotTest() {
+
+  private lateinit var mActivity: AppCompatActivity
+
+  @get:Rule
+  var activityRule: ActivityTestRule<MainActivity> =
+    ActivityTestRule(MainActivity::class.java)
+
+  @Before
+  fun setup() {
+    setDimensions(700, 400)
+    mActivity = activityRule.activity
+  }
+
+  @Test
+  fun screenshotTestCalendarDefault() {
+    val calendar = BpkCalendar(testContext)
+    val controller = BpkCalendarControllerImpl(
+      false,
+      Locale.UK,
+      getDate(2019, 0, 2),
+      getDate(2019, 11, 31))
+
+    calendar.setController(controller)
+    snap(wrapWithBackground(calendar))
+  }
+
+  @Test
+  fun screenshotTestCalendarWithStartDateSelected() {
+    val calendar = BpkCalendar(testContext)
+    val controller = BpkCalendarControllerImpl(
+      false,
+      Locale.UK,
+      getDate(2019, 0, 2),
+      getDate(2019, 11, 31))
+
+    calendar.setController(controller)
+    val wrapped = wrapWithBackground(calendar)
+
+    val asyncScreenshot = prepareForAsyncTest()
+
+    mActivity.runOnUiThread {
+      val rootLayout = mActivity.findViewById(android.R.id.content) as FrameLayout
+      rootLayout.addView(wrapped)
+    }
+
+    Espresso.onData(CoreMatchers.anything())
+      .atPosition(0)
+      .perform(ViewActions.click())
+      .check { _, _ ->
+        setupView(wrapped)
+        asyncScreenshot.record(wrapped)
+      }
+  }
+
+  @Test
+  fun screenshotTestCalendarWithSameStartAndEndDateSelected() {
+    val calendar = BpkCalendar(testContext)
+    val controller = BpkCalendarControllerImpl(
+      false,
+      Locale.UK,
+      getDate(2019, 0, 2),
+      getDate(2019, 11, 31))
+
+    calendar.setController(controller)
+    val wrapped = wrapWithBackground(calendar)
+
+    val asyncScreenshot = prepareForAsyncTest()
+
+    mActivity.runOnUiThread {
+      val rootLayout = mActivity.findViewById(android.R.id.content) as FrameLayout
+      rootLayout.addView(wrapped)
+    }
+
+    Espresso.onData(CoreMatchers.anything())
+      .atPosition(0)
+      .perform(ViewActions.click())
+
+    Espresso.onData(CoreMatchers.anything())
+      .atPosition(0)
+      .perform(ViewActions.click())
+      .check { _, _ ->
+        setupView(wrapped)
+        asyncScreenshot.record(wrapped)
+      }
+  }
+
+  @Test
+  fun screenshotTestCalendarWithStartAndEndDateSelected() {
+    val calendar = BpkCalendar(testContext)
+    val controller = BpkCalendarControllerImpl(
+      false,
+      Locale.UK,
+      getDate(2019, 0, 2),
+      getDate(2019, 11, 31))
+
+    calendar.setController(controller)
+    val wrapped = wrapWithBackground(calendar)
+
+    val asyncScreenshot = prepareForAsyncTest()
+
+    mActivity.runOnUiThread {
```

Comment:
Minor nit, but I prefer we don't use the old Android code style but rather follow the new Google Kotlin style guide. The Hungarian Notation is generally discouraged even by Google now.

btw I'm referring to `mActivity` here

Reply:
Agreed, I copied this and I forgot to change :). Will do